core/services/ocr2/models/models.go

```diff
@@ -5,3 +5,7 @@ type MercuryCredentials struct {
 	Username string
 	Password string
 }
+
+func (mc *MercuryCredentials) Validate() bool {
+	return mc.URL != "" && mc.Username != "" && mc.Password != ""
+}
```

Comment:
are we calling this somewhere? Will this fail if credentials haven't been provided?

Reply:
yes. it's used in `doCheck` function in `registry.go`. it's only checked if the registry mercury lookup feature is turned on.